wayland-client/src/event_queue.rs

```diff
@@ -730,3 +730,47 @@ macro_rules! delegate_dispatch {
         )*
     };
 }
+
+/// A helper macro which delegates a set of [`Dispatch`] implementations for proxies to a static handler.
+///
+/// # Usage
+///
+/// This macro is useful to implement [`Dispatch`] for interfaces where events are unimportant to
+/// the current application and can be ignored.
+///
+/// # Example
+///
+/// ```
+/// use wayland_client::{delegate_noop, protocol::{wl_data_offer, wl_subcompositor}};
+///
+/// /// The application state
+/// struct ExampleApp {
+///     // ...
+/// }
+///
+/// // Ignore all events for this interface:
+/// delegate_noop!(ExampleApp: [wl_data_offer::WlDataOffer: {}]);
+///
+/// // This interface should not emit events:
+///
+/// delegate_noop!(ExampleApp: [wl_subcompositor::WlSubcompositor: unreachable!() ]);
+/// ```
+#[macro_export]
+macro_rules! delegate_noop {
+    ($dispatch_from: ty: [ $($interface: ty: $expr: expr),* $(,)?]) => {
+        $(
+            impl $crate::Dispatch<$interface, ()> for $dispatch_from {
+                fn event(
+                    _: &mut Self,
+                    _: &$interface,
+                    _: <$interface as $crate::Proxy>::Event,
+                    _: &(),
+                    _: &$crate::Connection,
+                    _: &$crate::QueueHandle<Self>,
+                ) {
+                    $expr
```

Comment:
I'm not fan of the way this still allows basically any expression to be put here, that's an open road to having unwanted uses of that macro.

Instead and given we'd want to align this on https://github.com/Smithay/wayland-rs/pull/532 , I'd drop the ability to delegate multiple interface in the same call, and change the macro so that its invocation looks something like this:

```rust
// For an interface with no event, examples to `unreachable!()`
delegate_noop!(State: WlCompositor);
// For when you explicitly want to ignore events
delegate_noop!(State: ignore WlCompositor);
```

Reply:
I like this approach a lot better. Lemme try and figure out how to parse the "ignore" string in a macro rule.

Reply:
I'm not sure how to pares the `ignore` part of the macro. The Rust docs just say "This section is incomplete" for literals, and I can't find any examples of this being used. Do you know of any? Do you have any pointers for me? I'm kind stuck on that bit.

Reply:
AFAIK something like this should work:

```rust
macro_rules! delegate_noop {
    ($dispatch_from: ty : ignore $interface:ty) => {
        /* ... */
    };
    ($dispatch_from: ty : $interface:ty) => {
        /* ... */
    };
}
```

Reply:
Thanks, that was much simpler than I thought!

I think the latest PR version should match all expectations.

Reply:
Not completely, it's still missing the adjustments to allow for generics in the `delegate_to` type, like was done in #532. I'd prefer all such macros follow the same convention. As said earlier, doing this will also mean removing the ability to delegate multiple interfaces in a single macro call.

Reply:
I don't quite follow, the usage is identical to that of `delegate_dispatch`. Even with #532, you can still pass multiple interfaces to the macro.

Reply:
Even the example does that: https://github.com/smithay/wayland-rs/blob/8ca0a9b2b390957f15c237bf199b8479148f30ea/wayland-client/src/event_queue.rs#L765

Reply:
Never mind, the example is broken, test is `ignore`. I'll fix that too.

Reply:
Oh yeah, my error, I didn't notice the PR that modified the macro didn't update the example. :(

Reply:
See https://github.com/Smithay/wayland-rs/pull/550